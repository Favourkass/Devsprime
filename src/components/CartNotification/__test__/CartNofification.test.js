import renderer from 'react-test-renderer'; 
import React from "react";
import ReactDOM from "react-dom";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import CartNofication from "./../CartNotification";

afterEach(cleanup);

it("renders without crashing", ()=>{
  const div = document.createElement("div");
  ReactDOM.render("<CartNotification />", div)
  ReactDOM.unmountComponentAtNode(div)
});

it("renders cart correctly", ()=>{
  const {getByTestId} = render(<CartNofication number="3" />)
  expect(getByTestId("button")).toHaveTextContent("3")
});

it("renders cart correctly", ()=>{
  const {getByTestId} = render(<CartNofication number="7" />)
  expect(getByTestId("button")).toHaveTextContent("7")
});
