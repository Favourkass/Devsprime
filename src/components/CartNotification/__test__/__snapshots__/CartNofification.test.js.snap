// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdnxRM eBFDtt"
  data-testid="button"
>
  <img
    alt="cart"
    className="sc-hKFxyN lQzRk"
    src="cart.png"
  />
  <div
    className="sc-gtsrHT hYZwfB"
  >
    <p
      className="sc-dlnjwi ieBAV"
    >
      7
    </p>
  </div>
</div>
`;
